### DISCUSSION TITLE
about unix stream

### DISCUSSION THREAD

**Discussion by xuquan316** [curl/curl#12528] [Q&A] [answered]
### I did this

std::list<CURL*> curl_list;
for(int i = 0; i < 30；++i){
     CURL* curl = curl_easy_init();
      ......
     curl_list.push_back(curl);
}

for(CURL* elem:curl_list){
        .....
        curl_easy_perform(elem);
        ....
}

### I expected the following

The code executes and creates a TCP connection and also creates more unix stream connections

![image](https://github.com/curl/curl/assets/8208277/5401d59e-7fb9-4e74-997f-beb1205ace10)

Is there a way to reduce the unix stream connection?

### curl/libcurl version

curl 8.4.0

### operating system

ubuntu 1804

**Comment by bagder** [comment, marked as answer]
The "stream" you see is a socketpair, used for IPC with the name resolver thread. There is one per thread, and there is one (short-lived) thread per name resolve, which typically happens in the beginning of a transfer.

In 8.5.0, libcurl uses a pipe instead of socketpair on most platforms.

**Reply by xqcool** [reply]
thanks，ill try using 8.5.0
